atoma-state/src/state_manager.rs

```diff
@@ -2910,14 +2910,11 @@ impl AtomaState {
     /// }
     /// ```
     #[instrument(level = "trace", skip(self))]
-    pub async fn is_api_token_valid(&self, user_id: i64, api_token: &str) -> Result<bool> {
-        let is_valid = sqlx::query(
-            "SELECT EXISTS(SELECT 1 FROM api_tokens WHERE user_id = $1 AND token = $2)",
-        )
-        .bind(user_id)
-        .bind(api_token)
-        .fetch_one(&self.db)
-        .await?;
+    pub async fn is_api_token_valid(&self, api_token: &str) -> Result<bool> {
+        let is_valid = sqlx::query("SELECT EXISTS(SELECT 1 FROM api_tokens WHERE token = $1)")
+            .bind(api_token)
```

Comment:
why don't you need the user_id anymore ?

Reply:
I'm relying on the tokens to be not equal, and I don't need any verification with the user_id (it's needed only on the jwt tokens)